### PR TITLE
Distplot fix

### PR DESCRIPTION
**Pull request by yankev**
Fixes issue #459 
screen shot 2016-05-20 at 2 45 18 pm

### CODE REVIEW

**Comment by yankev** [comment]
Going to include a keyword argument to swap between the new way (where histnorm = 'probability density'), and the original (histnorm = 'probability')

**Comment by yankev** [comment]
@kully @cldougl @theengineear @empet
Added an extra parameter called `histnorm` to deal with the two different types of distplots (the seaborn-like method/the original and probability density method)
I've kept the default for `histnorm` as `probability` for backwards compatibility.

File: plotly/tools.py [hunk 1]
```diff
@@ -4019,7 +4019,7 @@ def create_candlestick(open, high, low, close,
     @staticmethod
     def create_distplot(hist_data, group_labels,
                         bin_size=1., curve_type='kde',
-                        colors=[], rug_text=[],
+                        colors=[], rug_text=[], histnorm='probability',
                         show_hist=True, show_curve=True,
                         show_rug=True):
         """
```
Comment (cldougl) [inline plotly/tools.py]: @yankev can you add that param to the docstring as well (so around line 4039)

File: plotly/tools.py [hunk 2]
```diff
@@ -4019,7 +4019,7 @@ def create_candlestick(open, high, low, close,
     @staticmethod
     def create_distplot(hist_data, group_labels,
                         bin_size=1., curve_type='kde',
-                        colors=[], rug_text=[],
+                        colors=[], rug_text=[], histnorm='probability',
```
Comment (cldougl) [inline plotly/tools.py]: @yankev can you add that param to the docstring as well (so around line 4039)

**Comment by empet** [comment]
@yankev The seaborn  distplot is equivalent to plotly distplot corresponding to `histnorm=probability density`.  Leaving the default as `probability` it's not a good idea because a lot of plots on Plotly cloud used wrong this key and people that take as examples these older plots can think that this version leads to a histogram comparable with an estimate of the probbaility density associated to data.

**Comment by empet** [comment]
Here is the proof that seaborn distplot works like Plotly distplot with `histnorm=probability density` (see the last plot):[http://nbviewer.jupyter.org/gist/empet/1bf38aea056f1d60481e600fc68c21be?flush_cache=true](http://nbviewer.jupyter.org/gist/empet/1bf38aea056f1d60481e600fc68c21be?flush_cache=true)

**Comment by yankev** [comment]
@empet oh I see, thanks. Initially I was trying to keep some backwards compatibility, but it appears as though setting `histnorm = 'probability density'` is the correct approach.

**Comment by empet** [comment]
@yankev Yes, this is the recommended default value for histnorm in distplot.

File: plotly/tools.py [hunk 3]
```diff
@@ -5108,7 +5111,9 @@ def make_kde(self):
             self.curve_y[index] = (scipy.stats.gaussian_kde
                                    (self.hist_data[index])
                                    (self.curve_x[index]))
-            self.curve_y[index] *= self.bin_size[index]
+
+            if self.histnorm == 'probability':
```
Comment (theengineear) [inline plotly/tools.py]: Can you pull these into constants at the top of the file? Hard-coded things like this always raise 🚨's for me.

**Comment by theengineear** [comment]
@yankev tests look great, thanks for adding those! 💃 me after pulling those `histnorm` values out into some constants somewhere!

**Pull request merged**